pages/api/level/image/[id].ts

```diff
@@ -0,0 +1,160 @@
+import * as PImage from 'pureimage';
+import { LevelImageModel, LevelModel } from '../../../../models/mongoose';
+import { NextApiRequest, NextApiResponse } from 'next';
+import LevelDataType from '../../../../constants/levelDataType';
+import { ObjectId } from 'bson';
+import { PassThrough } from 'stream';
+import dbConnect from '../../../../lib/dbConnect';
+import getTs from '../../../../helpers/getTs';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method === 'GET') {
+    if (!req.query) {
+      res.status(400).send('Missing required parameters');
+
+      return;
+    }
+
+    const { id } = req.query;
+
+    if (!id) {
+      return res.status(400).json({
+        error: 'Missing required parameters',
+      });
+    }
+
+    await dbConnect();
+    const levelImage = await LevelImageModel.findOne({ levelId: id });
+
+    if (levelImage) {
+      res.setHeader('Content-Type', 'image/png');
+      res.setHeader('Content-Length', levelImage.image.length);
+      res.status(200).send(levelImage.image);
+
+      return;
+    }
+
+    const level = await LevelModel.findOne({
+      _id: id,
+    });
+
+    if (!level) {
```

Comment:
This check should probably be above the level image query- otherwise we are querying unnecessarily and also added the wrong header for json request